src/apm/php.go

```diff
@@ -45,52 +62,57 @@ func InjectPhpagent(phpSpec v1alpha1.Php, pod corev1.Pod, index int) (corev1.Pod
 		}
 	}
 
-	const (
-		phpConcatEnvValues = false
-		concatEnvValues    = true
-	)
-
-	setPhpEnvVar(container, envPhpsymbolicOption, phpSymbolicOptionArgument, phpConcatEnvValues)
-
-	setPhpEnvVar(container, envPhpSilentOption, phpSilentOptionArgument, phpConcatEnvValues)
-
-	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+	// define instrumentation artifact volume
+	instVolume := corev1.VolumeMount{
 		Name:      volumeName,
 		MountPath: "/newrelic-instrumentation",
-	})
+	}
+
+	setPhpEnvVar(container, envIniScanDirKey, envIniScanDirVal, true)
+
+	container.VolumeMounts = append(container.VolumeMounts, instVolume)
 
-	// We just inject Volumes and init containers for the first processed container.
+	// We only inject Volumes and init containers for the first processed container.
 	if isInitContainerMissing(pod) {
+		initContainer := corev1.Container{
+			Name:    initContainerName,
+			Image:   phpSpec.Image,
+			Command: []string{"/bin/sh"},
+			Args: []string{
+				"-c", "cp -a /instrumentation/. /newrelic-instrumentation/ && /newrelic-instrumentation/k8s-php-install.sh " + apiNum + " && /newrelic-instrumentation/nr_env_to_ini.sh",
+			},
+			VolumeMounts: []corev1.VolumeMount{instVolume},
+		}
+
+		// inject the spec env vars into the initcontainer in order to construct the ini file
+		initContainer.Env = append(initContainer.Env, container.Env...)
+
+		// inject the license key secret
+		optional := true
+		initContainer.Env = append(initContainer.Env, corev1.EnvVar{
+			Name: "NEW_RELIC_LICENSE_KEY",
+			ValueFrom: &corev1.EnvVarSource{
```

Comment:
Is it possible to reach this point and the secret for the license key hasn't been defined or does the agent operator not allow this state on its startup?

Reply:
I don't believe so. Part of the installation instructions include defining the license key in such a way that it's guaranteed to be picked up by all agents by the point the `Inject{language}agent` function is called. Anecdotally this works perfectly when following the install guide in my local testing.